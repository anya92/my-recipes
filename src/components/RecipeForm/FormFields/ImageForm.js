import React, { Component } from 'react';
import styled from 'styled-components';

import Label from '../styled/Label';
import Input from '../styled/Input';

const Image = styled.div`
  height: 300px;
  background-color: #EEE;
  margin-bottom: 10px;
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;

const ChangeImageLink = styled.div`
  font-size: 1.1rem;
  cursor: pointer;
  color: dodgerblue;
  text-align: center;
  background-color: #FFF;  
`;

const StyledImageForm = styled.div`
  margin-bottom: 10px;
  div {
    display: flex;
    span {
      background-color: dodgerblue;
      color: #FFF;
      display: inline-flex;
      align-items: center;
      padding: 0 10px;
      border-left: 2px solid #FFF;
      cursor: pointer;
    }
  }
`;

class ImageForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      image: this.props.image,
      showImage: this.props.image !== ''
    };
  }

  handleImageChange = e => {
    const image = e.target.value;
    this.setState(() => ({ image }));
  }

  setImage = () => {
    const { image } = this.state;
    this.props.setImage(image);
    this.setState(() => ({ showImage: true }));
  }

  render() {
    if (this.state.showImage) {
      return (
        <div>
          <Image>
            <img src={this.state.image} alt="" />
          </Image>  
          <ChangeImageLink onClick={() => this.setState({ showImage: false })}>Change photo</ChangeImageLink>
        </div>
      );
    }
    return (
      <StyledImageForm>
        <Image>
          {this.state.image && <img src={this.state.image} alt="" />}
        </Image>
        <Label htmlFor="image">Add photo</Label>
        <div>
          <Input
            type="text"
            id="image"
            placeholder="https://"
            onChange={this.handleImageChange}
            value={this.state.image}
          />
          <span onClick={this.setImage}>Save</span>
        </div>
      </StyledImageForm>
    );
  }
}

export default ImageForm;
