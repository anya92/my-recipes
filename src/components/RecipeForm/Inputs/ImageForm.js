import React, { Component } from 'react';
import styled from 'styled-components';

const Image = styled.div`
  height: 300px;
  background-color: #EEE;
  margin-bottom: 10px;
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;

const ChangeImageLink = styled.div`
  font-size: 1.1rem;
  cursor: pointer;
  color: dodgerblue;
  text-align: center;
  background-color: #FFF;  
`;

const StyledImageForm = styled.div`
  margin-bottom: 10px;
  label {
    display: block;
    color: #4F4F4F;
    font-weight: 700;
    margin: 5px 0;
  }
  div {
    display: flex;
    span {
      background-color: dodgerblue;
      color: #FFF;
      display: inline-flex;
      align-items: center;
      padding: 0 10px;
      border-left: 2px solid #FFF;
      cursor: pointer;
    }
  }
  input {
    width: 100%;
    border: none;
    background-color: #F2F2F2;
    padding: 8px 2px;
    border-bottom: 2px solid #F2F2F2;
    &:focus {
      border-bottom-color: dodgerblue;
      outline: none;
    }
  }
`;

class ImageForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      image: this.props.image,
      showImage: this.props.image !== ''
    };
  }

  handleImageChange = e => {
    const image = e.target.value;
    this.setState(() => ({ image }));
  }

  setImage = () => {
    const { image } = this.state;
    this.props.setImage(image);
    this.setState(() => ({ showImage: true }));
  }

  render() {
    if (this.state.showImage) {
      return (
        <div>
          <Image>
            <img src={this.state.image} alt="" />
          </Image>  
          <ChangeImageLink onClick={() => this.setState({ showImage: false })}>Change photo</ChangeImageLink>
        </div>
      );
    }
    return (
      <StyledImageForm>
        <Image>{this.state.image && <img src={this.state.image} alt="" /> }</Image>
        <label htmlFor="image">Add photo</label>
        <div>
          <input type="text" id="image" placeholder="https://" onChange={this.handleImageChange} value={this.state.image} />
          <span onClick={this.setImage}>Save</span>
        </div>
      </StyledImageForm>
    );
  }
}

export default ImageForm;
