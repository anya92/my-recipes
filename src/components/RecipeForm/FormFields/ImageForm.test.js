import React from 'react';
import { shallow } from 'enzyme';

import ImageForm from './ImageForm';

let image;
let setImage;
let wrapper;

beforeEach(() => {
  image = '';
  setImage = jest.fn();
  wrapper = shallow(<ImageForm image={image} setImage={setImage} />);
});


test('should render ImageForm with no image correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ImageForm with image correctly', () => {
  image = 'assets/images/pancakes.jpg';
  wrapper = shallow(<ImageForm image={image} setImage={setImage} />);
  expect(wrapper).toMatchSnapshot();
});

test('should set image on input change', () => {
  const value = 'assets/images/nuts.jpg';
  wrapper.find('Input').simulate('change', {
    target: { value },
  });
  expect(wrapper.state('image')).toBe(value);
});

test('should hide image on ChangeImage click', () => {
  image = 'assets/images/pancakes.jpg';
  wrapper = shallow(<ImageForm image={image} setImage={setImage} />);
  expect(wrapper.state('showImage')).toEqual(true);
  wrapper.find('ChangeImage').simulate('click');
  expect(wrapper.state('showImage')).toEqual(false);
});

test('should call setImage and show the image on "Save" span click', () => {
  expect(wrapper.state('showImage')).toEqual(false);
  wrapper.find('span').simulate('click');
  expect(setImage).toHaveBeenLastCalledWith('');
  expect(wrapper.state('showImage')).toEqual(true);
});
