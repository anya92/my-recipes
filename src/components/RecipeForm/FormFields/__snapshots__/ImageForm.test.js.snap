// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ImageForm with image correctly 1`] = `
<div>
  <styled.div>
    <img
      alt=""
      src="assets/images/pancakes.jpg"
    />
  </styled.div>
  <ChangeImage
    onClick={[Function]}
  >
    Change photo
  </ChangeImage>
</div>
`;

exports[`should render ImageForm with no image correctly 1`] = `
<styled.div>
  <styled.div />
  <styled.label
    htmlFor="image"
  >
    Add photo
  </styled.label>
  <div>
    <Input
      id="image"
      onChange={[Function]}
      placeholder="https://"
      type="text"
      value=""
    />
    <span
      onClick={[Function]}
    >
      Save
    </span>
  </div>
</styled.div>
`;
